src/app/new-courses-page/new-courses-page.component.ts

```diff
@@ -1,60 +1,76 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
 
-import { CoursesService } from '@shared/services';
 import { Course } from '@app/models';
 
-import { Observable, Subject } from 'rxjs';
+import { Subject } from 'rxjs';
 import { takeUntil } from 'rxjs/operators';
 
+import { Store, select } from '@ngrx/store';
+import { selectEditedData } from '@app/core/@ngrx';
+import * as CoursesActions from '@app/core/@ngrx/courses/courses.actions';
+
 @Component({
   selector: 'app-new-courses-page',
   templateUrl: './new-courses-page.component.html',
   styleUrls: ['./new-courses-page.component.scss']
 })
 export class NewCoursesPageComponent implements OnInit, OnDestroy {
-  courseId: string;
-  
-  course$: Observable<Course>;
+  course: Course;
 
-  private unsubscribe: Subject<void> = new Subject();
+  private unsubscribe$: Subject<void> = new Subject();
 
   constructor(
-    private coursesService: CoursesService,
+    private store: Store,
     private router: Router,
     private route: ActivatedRoute,
   ) {}
 
   ngOnInit(): void {
-    this.courseId = this.route.snapshot.paramMap.get('id');
-
-    if (this.courseId != null) {
-      this.course$ = this.coursesService.getCourse(Number(this.courseId));
-    }
+      let observer: any = {
+        next: (course: Course) => {
+          if (course) {
+            this.course = { ...course };
+          }
+        },
+        error(err) {
+          console.log(err); 
+        },
+        complete() {
+          console.log('Stream is completed');
+        }
+      };
+  
+      this.store
+        .pipe(
+          select(selectEditedData),
+          takeUntil(this.unsubscribe$)
+        )
+        .subscribe(observer);
+  
+      observer = {
```

Comment:
same here. It's a bad practice to reuse same observable. Create a new subscription as you would do with rxjs